libs/pika/command_line_handling/tests/unit/get_env_var_as.cpp

```diff
@@ -0,0 +1,56 @@
+//  Copyright (c) 2023 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <pika/config.hpp>
+#include <pika/command_line_handling/get_env_var_as.hpp>
+#include <pika/init.hpp>
+#include <pika/testing.hpp>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+int pika_main(int /*argc*/, char* /*argv*/[])
+{
+    float f = pika::detail::get_env_var_as<float>("float", 0.0);
+    PIKA_TEST_EQ(f, 3.1415f);
+
+    double d = pika::detail::get_env_var_as<double>("double", 0.0);
+    PIKA_TEST_EQ(d, 6.28);
+
+    std::int32_t i32 = pika::detail::get_env_var_as<std::int32_t>("int32", 0);
+    PIKA_TEST_EQ(i32, -65536);
+
+    std::int64_t i64 = pika::detail::get_env_var_as<std::int64_t>("int64", 0);
+    PIKA_TEST_EQ(i64, -123465536);
+
+    std::uint32_t u32 = pika::detail::get_env_var_as<std::uint32_t>("uint32", 0);
+    PIKA_TEST_EQ(u32, 65536u);
+
+    std::uint64_t u64 = pika::detail::get_env_var_as<std::uint64_t>("uint64", 0);
+    PIKA_TEST_EQ(u64, 123465536u);
+
+    std::string s = pika::detail::get_env_var_as<std::string>("string", "wrong");
+    PIKA_TEST_EQ(s, std::string("hello-world"));
+
+    return pika::finalize();
+}
+///////////////////////////////////////////////////////////////////////////////
+int main(int argc, char* argv[])
+{
+    putenv(const_cast<char*>("float=3.1415"));
```

Comment:
This is apparently not standard C or C++, but posix. Not asking you to change anything (major) but could you add a comment about it and `#include <stdlib.h>` (https://www.man7.org/linux/man-pages/man3/putenv.3.html)?